views/js/runner/plugins/tools/textToSpeech/textToSpeech.js

```diff
@@ -0,0 +1,374 @@
+                }
+            })
+            .hide(); // Hide settings menu to begin
+
+            // Hide/show volume and speed sliders
+            $this.find('.settings > .settings-menu > .option')
+            .on('hover', function () {
+                $(this).find('.slider-container').show();
+            })
+            .on('mouseleave', function () {
+                $(this).find('.slider-container').hide();
+            });
+
+            // Hide slider to begin
+            $this.find('.settings > .settings-menu > .option > .slider-container').hide();
+
+            // Text to speech actions
+            $this.find('.click-to-speak').on('click', this.clickToSpeak);
+            $this.find('.play')          .on('click', this.play);
+            $this.find('.pause')         .on('click', this.pause)       .hide();
+            $this.find('.stop')          .on('click', this.stop);
+
+            // Settings menu's volume slider action
+            $this.find('.settings > .settings-menu > .volume .slider')
+            .noUiSlider({
+                animate: true,
+                connected: true,
+                range: {
+                    min: 0,
+                    max: 100
+                },
+                start: volume,
+                step: 10
+            })
+            .on('change', this.setVolume);
+
+            // Settings menu's speed slider
+            $this.find('.settings > .settings-menu > .speed .slider')
+            .noUiSlider({
+                animate: true,
+                connected: true,
+                range: {
+                    min: 0,
+                    max: 100
+                },
+                start: speed,
+                step: 10
+            })
+            .on('change', this.setSpeed);
+        })
+        .on('clickToSpeak', function () {
+            var $el = this.getElement();
+
+            if (options.enableClickToSpeak) {
+                $('.click-to-speak', $el).addClass('active');
+                $('.play', $el).addClass('disabled').show();
+                $('.pause', $el).addClass('disabled').hide();
+                options.$contentArea.css('cursor', 'pointer');
+            } else {
+                $('.click-to-speak', $el).removeClass('active');
+                $('.play', $el).removeClass('disabled').show();
+                $('.pause', $el).removeClass('disabled').hide();
+                options.$contentArea.css('cursor', 'default');
+            }
+        })
+        .on('play', function () {
+            var $el = this.getElement();
+
+            $('.play', $el).hide();
+            $('.pause', $el).show();
+        })
+        .on('pause', function () {
+            var $el = this.getElement();
+
+            $('.play', $el).show();
+            $('.pause', $el).hide();
+        })
+        .on('stop', function () {
+            var $el = this.getElement();
+
+            $('.play', $el).show();
+            $('.pause', $el).hide();
```

Comment:
Too many repetition of the same selector. Would be better to get them all in the sam variables like: 
```
$element = {
    play : this.getElement().find('.play'),
    pause : this.getElement().find('.pause'),
    clickToSpeak : this.getElement().find('.click-to-speak'),
    ...
}

Reply:
1) You inadvertently pointed at that it is better to use `find()` over a selector when you have the jquery element you want to search with already.

2) To your main point, you're correct that these jquery elements are used widely enough to justify an abstraction. I believe, however, that this is not an abstraction I want to create.
    - This would require placing these variables on the `ui/component` object. However, I don't want consumers of this object to ever consider these variables. 
    - Over abstraction can cause slowed development (this would be a good place to cite the article I read on this, but... sorry, it should be a quick google search away if interested). So, when such an instance occurs that abstraction provides little benefit (e.g. `$play` vs. `$el.find('.play')`), I tend not to. For instance, when bug finding, you wouldn't have to trust that the `$play` variable was created correctly.

This is obviously a debatable point, so you're welcome to disagree. But, for this case, I'd need you to support your request for making this change.